src/lyk.c

```diff
@@ -92,23 +92,28 @@ void set_location(Proto *f, int i) {
 }
 
 inline void yk_on_instruction_loaded(Proto *f, Instruction i, int idx) {
-  // YKOPT: Reallocating for every instruction is inefficient.
-  YkLocation **new_locations = calloc(f->sizecode, sizeof(YkLocation *));
-  lua_assert(new_locations != NULL && "Expected yklocs to be defined!");
+  if (f->yklocs == NULL || f->sizecode > f->yklocs_size) {
```

Comment:
I wonder if this should be something like:

```c
if (f->yklocs == NULL) {
  f->ykcallocs = calloc(...);
  f->yklocs_size = ...;
  ...
}

if (f->sizecode > f->yklocs_size) {
  realloc(...)
}
```

A related question: are `sizecode` and `yklocs_size` counting number of elements or number of bytes?

Reply:
my first implementation used realloc, but I hadn't thought of this. Is `realloc` always faster than `calloc`?
`sizecode` keeps track of number of instructions in the bytecode array, I am guessing that `yklocs_size` contains number of locations. So to answer your question, so both are counting number of elements.

Reply:
If you use `calloc` you allocate two chunks of memory, and have to free one. `realloc` does that *and* it moves the old data you had.

Reply:
changed here: 60dca65636d04e80a075efceee2a862a309bcf64

Reply:
> I am guessing that yklocs_size contains number of locations

Can you check and make sure of this please?

Reply:
it is number of locations. Here is the commit fixing the comments: 1dee9a94e2bc6c0170ee7b02afaaa8885736d963